types/yup/index.d.ts

```diff
@@ -642,8 +642,9 @@ type Id<T> = { [K in keyof T]: T[K] };
 type RequiredProps<T> = Pick<T, Exclude<keyof T, KeyOfUndefined<T>>>;
 type NotRequiredProps<T> = Partial<Pick<T, KeyOfUndefined<T>>>;
 type InnerInferType<T> =
-    | (T extends Array<infer T> ? T[] : Id<NotRequiredProps<T> & RequiredProps<T>>)
+    | (T extends Array<infer T> ? InnerInferTypeArray<T> : Id<NotRequiredProps<T> & RequiredProps<T>>)
     | PreserveOptionals<T>;
+interface InnerInferTypeArray<T> extends Array<InnerInferType<T>> {}
```

Comment:
Any reason this needs to be an interface? I think it would be clearer as `type InnerInferTypeArray<T> = Array<InnerInferType<T>>;`

Reply:
good question – it's because you can't use a type alias there; if you do, you get errors about type aliases not being able to recursively reference themselves. this limitation doesn't apply to interfaces.

Reply:
I was wondering about the reason for `interface InnerInferTypeArray<T>` in the fist place as it doesn't add any members from `Array<InnerInferType<T>>` so just using the laster should be the same. But I noticed compile errors with TypeScript@3.6 while TypeScript@3.7 has no compile errors.

LGTM